### ISSUE TITLE
Angular template: Cannot find the name 'Iterable', etc.  (Visual Studio Errors)

### ISSUE THREAD

**Issue by kakins** [aspnet/templating#459]
I just noticed this today, about 24 errors are now appearing in my Visual Studio error list.  I have not modified any of the default .ts files that came with the latest Angular template.

![image](https://user-images.githubusercontent.com/14915247/38456431-5d3148b0-3a4a-11e8-83f0-f2b50d1bff2e.png)

Included in these errors is one concerning a decorator that I've never encountered before.  Interestingly it's only in one file where this appears, even though the same decorator is used in multiple files:

![image](https://user-images.githubusercontent.com/14915247/38456465-95f4b268-3a4a-11e8-8766-5737650c93ef.png)

![image](https://user-images.githubusercontent.com/14915247/38456462-8c2cfee8-3a4a-11e8-8a47-3c4792d73b87.png)

**Comment by Eilon** [comment]
Hi @kakins , which version of the template are you using?

**Comment by Eilon** [comment]
Hi @kakins , which version of the template are you using?

**Comment by kakins** [comment]
This is the CLI-based template that I used following these instructions, using the dotnet cli:

https://docs.microsoft.com/en-us/aspnet/core/spa/angular?tabs=visual-studio

**Comment by Eilon** [comment]
@SteveSandersonMS - any idea on this?

**Comment by kakins** [comment]
I wish that were the case!  Unfortunately I have Node 9.5.0 & NPM 5.6.0.

**Comment by mkArtakMSFT** [comment]
@kakins, did you get this issue resolved? Maybe by updating the dotnet SDK or VS?

**Comment by mkArtakMSFT** [comment]
Hi. We're closing this issue as no response or updates have been provided in a timely manner and we have been unable to reproduce it. If you have more details and are encountering this issue please add a new reply and re-open the issue.

**Comment by Sl1MBoy** [comment]
I have the same issue. I have created a new angular project and there are no errors until I add any .ts with a decorator in it. For example I have added an empty module and I get this: 
![image](https://user-images.githubusercontent.com/29477710/41654654-6f7e5a42-748a-11e8-8294-78ee550254ea.png)

I get error for any decorator that I create, but no errors for the ones that came with the template.
Although the app works, these errors are annoying as you get more and more by creating new modules, components, input fields etc. 

Can someone help me what can cause this?

I am using:
`VS Community 2017 15.7.4

e:\>node -v
v8.11.2

e:\>npm -v
5.8.0

e:\>tsc -v
Version 2.7.2`

**Comment by Sl1MBoy** [comment]
Hi @mkArtakMSFT, can you reopen this please?

**Comment by kakins** [comment]
And I forgot to respond, but I do experience the same issue still.

**Comment by Eilon** [comment]
Thanks, we will evaluate this again in our next scan of bugs.

**Comment by Sl1MBoy** [comment]
@Eilon Do you have any update on this?

**Comment by Eilon** [comment]
@Sl1MBoy sorry unfortunately not yet.

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg, can you please look into this? Thanks!

**Comment by mkArtakMSFT** [comment]
@Sl1MBoy, can you please share with us the project you're facing these issues with?

**Comment by Sl1MBoy** [comment]
@mkArtakMSFT [https://github.com/Sl1MBoy/TestSpaTemplate](https://github.com/Sl1MBoy/TestSpaTemplate)

**Comment by ryanbrandenburg** [comment]
I believe your problem might be the version of tsc. I was briefly able to replicate when I had 2.7.2 installed (but not 2.7), but everything worked fine when I had 2.9.2. Try installing the latest version (I did so through npm).

**Comment by Sl1MBoy** [comment]
I have updated the tsc but it didn't solve the issue. However I think I have the solution. I saw in the properties of test.module.ts that it's package action is TypeScriptCompile, but all the other ts files have Build Action None. Then I have remembered that I created that file using VS context menu. After I have changed Build Action None for that file and restarted VS all the errors gone. So if you are creating the ts file using the angular cli you don't have this issue, but as soon you add it via VS and don't change the properties you have errors. @kakins Could you please check if this solution solves it for you too?

**Comment by mkArtakMSFT** [comment]
@vijayrkn, this may be something you'd like to act on:

>Then I have remembered that I created that file using VS context menu. After I have changed Build Action None for that file and restarted VS all the errors gone. So if you are creating the ts file using the angular cli you don't have this issue, but as soon you add it via VS and don't change the properties you have errors. @kakins Could you please check if this solution solves it for you too?

**Comment by vijayrkn** [comment]
@minestarks - Would you the be right person to look at this issue with VS Project -> Right click -> Add ts file?

**Comment by minestarks** [comment]
Yes, this sounds accurate and the behavior is by design. `TypeScriptCompile` is required to build TypeScript items using MSBuild, which is the default behavior, and the way to opt out of it (i.e. if you're using a separate system to compile TypeScript) is to set the item type back to `None`.
